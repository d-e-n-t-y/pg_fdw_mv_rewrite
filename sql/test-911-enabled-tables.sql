-- Anticipate rewrite
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;

-- Anticipate rewrite
set mv_rewrite.rewrite_enabled_for_tables = 'public.test';
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;

-- Anticipate no rewrite
set mv_rewrite.rewrite_enabled_for_tables = '';
DO LANGUAGE plpgsql $$ DECLARE t text; BEGIN FOR t IN
	explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key
LOOP IF t LIKE '%Rewritten%' THEN RAISE 'Rewritten'; END IF; END LOOP; END; $$;

-- Anticipate rewrite
set mv_rewrite.rewrite_enabled_for_tables = ' public.test, public.test   ,     public.test     ';
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;

-- Anticipate no rewrite
set mv_rewrite.rewrite_enabled_for_tables = 'public.not_test';
DO LANGUAGE plpgsql $$ DECLARE t text; BEGIN FOR t IN
	explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key
LOOP IF t LIKE '%Rewritten%' THEN RAISE 'Rewritten'; END IF; END LOOP; END; $$;

-- Anticipate no rewrite
set mv_rewrite.rewrite_enabled_for_tables = 'other_schema.test';
DO LANGUAGE plpgsql $$ DECLARE t text; BEGIN FOR t IN
	explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key
LOOP IF t LIKE '%Rewritten%' THEN RAISE 'Rewritten'; END IF; END LOOP; END; $$;

-- Anticipate message "MV rewrite not enabled for one or more table in the query"
set mv_rewrite.log_match_progress = 'true';
-- We don't actually care about the plan itself, only the log output.
DO LANGUAGE plpgsql $$ BEGIN EXECUTE 'explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;'; END; $$;

-- Anticipate rewrite
set mv_rewrite.log_match_progress = 'false';
set mv_rewrite.rewrite_enabled_for_tables to default;
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;
