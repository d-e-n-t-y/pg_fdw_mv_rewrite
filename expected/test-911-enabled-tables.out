-- Anticipate rewrite
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;
                 QUERY PLAN                 
--------------------------------------------
 Sort
   Output: key, (count(value))
   Sort Key: test.key
   ->  Custom Scan (MVRewriteScan)
         Output: key, (count(value))
         Rewritten: scan of public.test_mv1
(6 rows)

-- Anticipate rewrite
set mv_rewrite.rewrite_enabled_for_tables = 'public.test';
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;
                 QUERY PLAN                 
--------------------------------------------
 Sort
   Output: key, (count(value))
   Sort Key: test.key
   ->  Custom Scan (MVRewriteScan)
         Output: key, (count(value))
         Rewritten: scan of public.test_mv1
(6 rows)

-- Anticipate no rewrite
set mv_rewrite.rewrite_enabled_for_tables = '';
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;
                        QUERY PLAN                        
----------------------------------------------------------
 Finalize GroupAggregate
   Output: key, count(value)
   Group Key: test.key
   ->  Sort
         Output: key, (PARTIAL count(value))
         Sort Key: test.key
         ->  Gather
               Output: key, (PARTIAL count(value))
               Workers Planned: 2
               ->  Partial HashAggregate
                     Output: key, PARTIAL count(value)
                     Group Key: test.key
                     ->  Parallel Seq Scan on public.test
                           Output: key, value
(14 rows)

-- Anticipate rewrite
set mv_rewrite.rewrite_enabled_for_tables = ' public.test, public.test   ,     public.test     ';
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;
                 QUERY PLAN                 
--------------------------------------------
 Sort
   Output: key, (count(value))
   Sort Key: test.key
   ->  Custom Scan (MVRewriteScan)
         Output: key, (count(value))
         Rewritten: scan of public.test_mv1
(6 rows)

-- Anticipate no rewrite
set mv_rewrite.rewrite_enabled_for_tables = 'public.not_test';
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;
                        QUERY PLAN                        
----------------------------------------------------------
 Finalize GroupAggregate
   Output: key, count(value)
   Group Key: test.key
   ->  Sort
         Output: key, (PARTIAL count(value))
         Sort Key: test.key
         ->  Gather
               Output: key, (PARTIAL count(value))
               Workers Planned: 2
               ->  Partial HashAggregate
                     Output: key, PARTIAL count(value)
                     Group Key: test.key
                     ->  Parallel Seq Scan on public.test
                           Output: key, value
(14 rows)

-- Anticipate no rewrite
set mv_rewrite.rewrite_enabled_for_tables = 'other_schema.test';
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;
                        QUERY PLAN                        
----------------------------------------------------------
 Finalize GroupAggregate
   Output: key, count(value)
   Group Key: test.key
   ->  Sort
         Output: key, (PARTIAL count(value))
         Sort Key: test.key
         ->  Gather
               Output: key, (PARTIAL count(value))
               Workers Planned: 2
               ->  Partial HashAggregate
                     Output: key, PARTIAL count(value)
                     Group Key: test.key
                     ->  Parallel Seq Scan on public.test
                           Output: key, value
(14 rows)

-- Anticipate message "MV rewrite not enabled for one or more table in the query"
set mv_rewrite.log_match_progress = 'true';
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;
INFO:  mv_rewrite_add_rewritten_mv_paths: MV rewrite not enabled for one or more table in the query.
                        QUERY PLAN                        
----------------------------------------------------------
 Finalize GroupAggregate
   Output: key, count(value)
   Group Key: test.key
   ->  Sort
         Output: key, (PARTIAL count(value))
         Sort Key: test.key
         ->  Gather
               Output: key, (PARTIAL count(value))
               Workers Planned: 2
               ->  Partial HashAggregate
                     Output: key, PARTIAL count(value)
                     Group Key: test.key
                     ->  Parallel Seq Scan on public.test
                           Output: key, value
(14 rows)

-- Anticipate rewrite
set mv_rewrite.log_match_progress = 'false';
set mv_rewrite.rewrite_enabled_for_tables to default;
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test group by key order by key;
                 QUERY PLAN                 
--------------------------------------------
 Sort
   Output: key, (count(value))
   Sort Key: test.key
   ->  Custom Scan (MVRewriteScan)
         Output: key, (count(value))
         Rewritten: scan of public.test_mv1
(6 rows)

